src/backend/updates.rs

```diff
@@ -332,8 +305,18 @@ fn mint_credits() {
 
 #[export_name = "canister_update create_invite"]
 fn create_invite() {
-    let credits: Credits = parse(&arg_data_raw());
-    mutate(|state| reply(state.create_invite(caller(), credits)));
+    let (credits, credits_per_user, realm_id): (Credits, Option<Credits>, Option<RealmId>) =
+        parse(&arg_data_raw());
+    mutate(|state| reply(state.create_invite(caller(), credits, credits_per_user, realm_id)));
+}
+
+#[export_name = "canister_update update_invite"]
+fn update_invite() {
+    let (invite_code, credits, realm_id): (String, Option<Credits>, Option<RealmId>) =
+        parse(&arg_data_raw());
+    let user_id = read(|state| state.principal_to_user(caller()).expect("no user found").id);
```

Comment:
If I remember correctly this kind of app data is always extracted inside modules, not in `updates.rs` which mostly contains "plumbing" code.

Is there any reason to not simply pass `caller()` as a first argument and then find user and do all error handling inside `update_invite`?

Reply:
Fully agreed, I changed it.